### PR TITLE
Resource reload renames

### LINKED ISSUE #920: ResourceReloadListener#reload -> createReloadTask or something like that
`ResourceReloadListener#reload` is not supposed to perform any reloading on its own. Instead, its goal is to return a `CompletableFuture` which, when evaluated, is supposed to submit the actual reloading tasks to various `Executor`s.

If you try cheating the system by putting code directly in `reload` and returning some dummy `CompletableFuture` that does nothing, the game hangs when reloading, because the future returned by `Synchronizer#whenPrepared` never gets called.

Naming it something like `createReloadTask` seems like a way to discourage this path and clarify that no reloading should be done directly in this method.

### PR DESCRIPTION
**Pull request by liach**
Fixes #920

Signed-off-by: liach <liach@users.noreply.github.com>

### CODE REVIEW

**Comment by liach** [comment]
@LambdAurora Think you might be interested in this as you've worked on resource loader in the API.

**Review by LambdAurora — APPROVED** [review]
Loving this refactor.

Especially the javadoc, it's great!

File: mappings/net/minecraft/resource/ReloadableResourceManager.mapping [hunk 1]
```diff
@@ -1,13 +1,46 @@
...
 		ARG 3 packs
+			COMMENT a list of resource packs providing resources
 		ARG 4 initialStage
-	METHOD method_18232 beginMonitoredReload (Ljava/util/concurrent/Executor;Ljava/util/concurrent/Executor;Ljava/util/concurrent/CompletableFuture;Ljava/util/List;)Lnet/minecraft/class_4011;
+			COMMENT a completable future to be completed before this reload
+	METHOD method_18232 reload (Ljava/util/concurrent/Executor;Ljava/util/concurrent/Executor;Ljava/util/concurrent/CompletableFuture;Ljava/util/List;)Lnet/minecraft/class_4011;
+		COMMENT Performs a reload. Returns an object that yields some insights to the
+		COMMENT reload.
+		COMMENT
+		COMMENT <p>{@code prepareExecutor} may be asynchrous. {@code applyExecutor} must
```
Comment (liach) [inline mappings/net/minecraft/resource/ReloadableResourceManager.mapping]:
```suggestion
		COMMENT <p>{@code prepareExecutor} may be asynchronous. {@code applyExecutor} must
```

File: mappings/net/minecraft/resource/ResourceReloader.mapping [hunk 2]
```diff
@@ -1,30 +1,65 @@
...
-		ARG 4 initialStage
-	CLASS class_4047 Factory
-		METHOD create (Lnet/minecraft/class_3302$class_4045;Lnet/minecraft/class_3300;Lnet/minecraft/class_3302;Ljava/util/concurrent/Executor;Ljava/util/concurrent/Executor;)Ljava/util/concurrent/CompletableFuture;
-			ARG 1 helper
-			ARG 2 manager
-			ARG 3 listener
-			ARG 4 prepareExecutor
-			ARG 5 applyExecutor
+CLASS net/minecraft/class_3302 net/minecraft/resource/ResourceReloader
+	COMMENT A resource reloader performs actual reloading in a {@linkplain #reload
```
Comment (liach) [inline mappings/net/minecraft/resource/ResourceReloader.mapping]:
```suggestion
	COMMENT A resource reloader performs actual reloading in its {@linkplain #reload
```

File: mappings/net/minecraft/resource/ResourceReloader.mapping [hunk 3]
```diff
@@ -1,30 +1,65 @@
...
+	COMMENT reload} of a reloadable resource manager.
```
Comment (liach) [inline mappings/net/minecraft/resource/ResourceReloader.mapping]:
```suggestion
	COMMENT reload} in a reloadable resource manager it is registered to.
```

File: mappings/net/minecraft/resource/SinglePreparationResourceReloader.mapping [hunk 4]
```diff
@@ -0,0 +1,27 @@
+CLASS net/minecraft/class_4080 net/minecraft/resource/SinglePreparationResourceReloader
+	COMMENT A base resource reloader implementation that prepares an object in a
+	COMMENT single call (as opposed to in multiple concurrent tasks) and handles
+	COMMENT the prepared object in the apply stage.
+	COMMENT
+	COMMENT @param <T> the inermediate object type
```
Comment (liach) [inline mappings/net/minecraft/resource/SinglePreparationResourceReloader.mapping]:
```suggestion
	COMMENT @param <T> the intermediate object type
```

File: mappings/net/minecraft/resource/ResourceReloader.mapping [hunk 5]
```diff
@@ -1,30 +1,65 @@
...
-			ARG 5 applyExecutor
+CLASS net/minecraft/class_3302 net/minecraft/resource/ResourceReloader
+	COMMENT A resource reloader performs actual reloading in its {@linkplain #reload
+	COMMENT reload} in a reloadable resource manager it is registered to.
+	COMMENT
+	COMMENT @see ReloadableResourceManager
+	COMMENT @see SinglePreparationResourceReloader SinglePreparationResourceReloader
+	COMMENT (completes preparation in one method)
+	COMMENT @see SynchronousResourceReloader SynchronousResourceReloader
+	COMMENT (performs all reloading in the apply executor)
```
Comment (YanisBft) [inline mappings/net/minecraft/resource/ResourceReloader.mapping:9]: What happened here?

File: mappings/net/minecraft/resource/ResourceReloader.mapping [hunk 6]
```diff
@@ -1,30 +1,65 @@
...
+	METHOD method_22322 getName ()Ljava/lang/String;
+		COMMENT Returns a user-friendly name for logging.
+	METHOD method_25931 reload (Lnet/minecraft/class_3302$class_4045;Lnet/minecraft/class_3300;Lnet/minecraft/class_3695;Lnet/minecraft/class_3695;Ljava/util/concurrent/Executor;Ljava/util/concurrent/Executor;)Ljava/util/concurrent/CompletableFuture;
+		COMMENT Perform a reload. Returns a future that is completed when the reload
```
Comment (YanisBft) [inline mappings/net/minecraft/resource/ResourceReloader.mapping]:
```suggestion
		COMMENT Performs a reload. Returns a future that is completed when the reload
```

File: mappings/net/minecraft/resource/ResourceReloader.mapping [hunk 7]
```diff
@@ -1,30 +1,65 @@
...
+		COMMENT CompletableFuture#supplyAsync(Supplier, Executor)
+		COMMENT CompletableFuture.supplyAsync(..., prepareExecutor)}
+		COMMENT to ensure the prepare actions are done with the prepare executor. Then,
+		COMMENT you should have a completable future for all the prepared actions, and
+		COMMENT call {@linkplain CompletableFuture#thenCompose(Function)
+		COMMENT combinedPrepare.thenCompose(synchronizer::waitFor)}
+		COMMENT to notify the {@code synchronizer}. Finally, you should run {@linkplain
+		COMMENT CompletableFuture#thenAcceptAsync(Consumer, Executor)
+		COMMENT CompletableFuture.thenAcceptAsync(..., applyExecutor)} for apply actions.
+		COMMENT In the end, return the result of {@code thenAcceptAsync}.
```
Comment (YanisBft) [inline mappings/net/minecraft/resource/ResourceReloader.mapping]:
```suggestion
		COMMENT In the end, returns the result of {@code thenAcceptAsync}.
```

File: mappings/net/minecraft/resource/ResourceReloader.mapping (hunk 5 above)
Comment (liach) [reply mappings/net/minecraft/resource/ResourceReloader.mapping:9]: Line too long, wrapping. The rendered effect should be
`SynchronousResourceReloader (performs all reloading in the apply executor)`

**Review by YanisBft — APPROVED** [review]

**Comment by modmuss50** [comment]
This is an impactful refactor, needs more reviews.

**Comment by liach** [comment]
> 
> 
> This is an impactful refactor, needs more reviews.

Help advertise reviewing this pr on social media then :wink: so we can get more reviews

Also, some of the resource reloads is changed in 11a. If you plan to let me backport this please let me know.

**Review by Mysterious-Dev — APPROVED** [review]

**Review by jaskarth — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]
Im happy with these changes, however it will cause a few of fabric api's names to be a little odd now.

**Comment by liach** [comment]
It does. I thought this would be acceptable as we will push forward https://github.com/FabricMC/fabric/issues/1373 as resource api v1, in which we probably will support building a default data pack at runtime than at compile time (because at compile time, the other mods aren't known.). In that we probably will include parts like shedaniel's conditional resource.

**Pull request merged**